core/src/main/scala/io/udash/properties/seq/ForwarderReadableSeqProperty.scala

```diff
@@ -9,97 +9,79 @@ private[properties] trait ForwarderReadableSeqProperty[A, B, ElemType <: Readabl
 
   protected def origin: ReadableSeqProperty[A, OrigType]
 
-  private var originListenerRegistration: Registration = _
+  protected final var transformedElements: MBuffer[ElemType] = CrossCollections.createArray[ElemType]
   private var originStructureListenerRegistration: Registration = _
 
-  protected final def initialized: Boolean = originListenerRegistration != null && originListenerRegistration.isActive
+  protected final def initialized: Boolean =
+    originStructureListenerRegistration != null && originStructureListenerRegistration.isActive
+
+  protected def getFromOrigin(): BSeq[B]
+  protected def transformElements(elemProperties: BSeq[OrigType]): BSeq[ElemType]
+  protected def transformPatchAndUpdateElements(patch: Patch[OrigType]): Opt[Patch[ElemType]]
+
+  protected def originStructureListener(patch: Patch[OrigType]): Unit =
+    transformPatchAndUpdateElements(patch).foreach { transformed =>
+      fireElementsListeners(transformed)
+      valueChanged()
+    }
+
+  protected def onListenerInit(originElems: BSeq[OrigType]): Unit = {
+    transformedElements = CrossCollections.toCrossArray(transformElements(originElems))
+  }
 
-  protected def originListener(originValue: BSeq[A]): Unit = {}
-  protected def originStructureListener(patch: Patch[OrigType]): Unit = {}
-  protected def onListenerInit(): Unit = {}
   protected def onListenerDestroy(): Unit = {}
 
-  protected def initOriginListeners(): Unit =
+  private def initOriginListener(): Unit =
     if (!initialized) {
       listeners.clear()
-      onListenerInit()
-      originListenerRegistration = origin.listen(originListener)
+      onListenerInit(origin.elemProperties)
       structureListeners.clear()
       originStructureListenerRegistration = origin.listenStructure(originStructureListener)
     }
 
-  protected def killOriginListeners(): Unit =
+  private def killOriginListener(): Unit =
     if (initialized && listeners.isEmpty && structureListeners.isEmpty) {
-      originListenerRegistration.cancel()
       onListenerDestroy()
-      originListenerRegistration = null
       originStructureListenerRegistration.cancel()
       originStructureListenerRegistration = null
     }
 
+  override def get: BSeq[B] =
+    if (initialized) transformedElements.map(_.get)
+    else getFromOrigin()
```

Comment:
I'm wondering whether this access should actually `initialize`, what do you think?

Reply:
We'd have no way to ever clean the listeners up :( I'm afraid it's something we have to live with in this design.